int32[] step_numbers
string[] step_descriptions
